### ISSUE TITLE
Make TOC available separately from content

### ISSUE THREAD

**Issue by matthiasrohmer** [grow/grow#858]
Currently the table of content is not available standalone and always ends up in a big string of rendered markdown. This doesn't really allow for more sophisticated page layouts where the TOC should for example reside in a sticky sidebar. Of course if someone doesn't want to rely on ugly `position: fixed;` hacks 😉 

To solve this it would be nice if maybe `documents.document_format.MarkdownDocumentFormat` would use a instance of Markdown and then for example `self.markdown.convert()` instead of `markdown.markdown()`. This would make the TOC available like this:

```jinja2
<div style="background-color: red;">
{{ doc.format.markdown.toc }}
</div>

<hr>

<div>
{{ doc.html|render }}
</div>
```

/edit: Sorry, misread the docs for the TOC extension.

**Comment by matthiasrohmer** [comment]
I've implemented this in a fork which would be ready to pull: [grow/documents/document_format.py#L195](https://github.com/matthiasrohmer/grow/blob/d8300217d5a65465144772000e8fcbe2279758f8/grow/documents/document_format.py#L195)

What I am not really happy with is that this breaks the pattern of all DocumentFormats sharing the same API. An alternative would be to return an Object from `formatted` that has the rendered markdown as string representation and a attribute `toc` to pull the TOC from. I am curious what you think.

**Comment by Zoramite** [comment]
So, in the doc instead of using the `[toc]` you would use the doc format to place the TOC where you want it in the view? I like the separation that would bring. Let me think things over a bit more and we can do a pull request.

**Comment by matthiasrohmer** [comment]
Yep, that's what we are trying to achieve. Let me know when you are ready to pull or have another solution in mind 🙂

**Comment by pbakaus** [comment]
@Zoramite friendly ping – if this was something you could look at in the next days, it would greatly help us launch the next iteration of ampproject.org :) thanks in advance!

**Comment by Zoramite** [comment]
Closed by #867
